src/main/java/org/xmldb/api/security/Permissions.java

```diff
@@ -0,0 +1,357 @@
+ * of permissions.
+ */
+public final class Permissions {
+  private static final char READ = 04;
+  private static final char WRITE = 02;
+  private static final char EXECUTE = 01;
+
+  private static final char SETID_CHAR = 's';
+  private static final char SETID_CHAR_NO_EXEC = 'S';
+  private static final char STICKY_CHAR = 't';
+  private static final char STICKY_CHAR_NO_EXEC = 'T';
+  private static final char READ_CHAR = 'r';
+  private static final char WRITE_CHAR = 'w';
+  private static final char EXECUTE_CHAR = 'x';
+  private static final char UNSET_CHAR = '-';
+  private static final char ALL_CHAR = 'a';
+  private static final char USER_CHAR = 'u';
+  private static final char GROUP_CHAR = 'g';
+  private static final char OTHER_CHAR = 'o';
+
+  public static final Pattern UNIX_SYMBOLIC_MODE_PATTERN =
+      Pattern.compile("((?:[augo]*(?:[+\\-=](?:[" + READ_CHAR + SETID_CHAR + STICKY_CHAR
+          + WRITE_CHAR + EXECUTE_CHAR + "])+)+),?)+");
+  public static final Pattern SIMPLE_SYMBOLIC_MODE_PATTERN = Pattern
+      .compile("(?:(?:" + READ_CHAR + "|" + UNSET_CHAR + ")(?:" + WRITE_CHAR + "|" + UNSET_CHAR
+          + ")(?:[" + EXECUTE_CHAR + SETID_CHAR + SETID_CHAR_NO_EXEC + "]|" + UNSET_CHAR
+          + ")){2}(?:" + READ_CHAR + "|" + UNSET_CHAR + ")(?:" + WRITE_CHAR + "|" + UNSET_CHAR
+          + ")(?:[" + EXECUTE_CHAR + STICKY_CHAR + STICKY_CHAR_NO_EXEC + "]|" + UNSET_CHAR + ")");
+
+  private Permissions() {}
+
+  private enum PermType {
+    READ, WRITE, EXECUTE, SETID, STICKY;
+  }
+
+  /**
+   * Parses the permissions from the given mode string.
+   * 
+   * The string can either be in one of two formats:
+   * <ol>
+   * <li>Unix Symbolic format as given to 'chmod' on Unix/Linux</li>
+   * <li>Simple Symbolic format e.g. "rwxr-xr-x"</li>
+   * </ol>
+   * 
+   * @param modeStr the mode to be parsed
+   * @return the set of parsed permissions
+   * @throws IllegalArgumentException if the given mode string is in the wrong format
+   */
+  public static Set<Permission> ofModeString(String modeStr) {
+    final Matcher simpleSymbolicModeMatcher = SIMPLE_SYMBOLIC_MODE_PATTERN.matcher(modeStr);
+    final EnumSet<Permission> permissions = EnumSet.noneOf(Permission.class);
+    if (simpleSymbolicModeMatcher.matches()) {
+      setSimpleSymbolicMode(permissions, modeStr);
+    } else {
+      final Matcher unixSymbolicModeMatcher = UNIX_SYMBOLIC_MODE_PATTERN.matcher(modeStr);
+      if (unixSymbolicModeMatcher.matches()) {
+        setUnixSymbolicMode(permissions, modeStr);
+      } else {
+        throw new IllegalArgumentException("Unknown mode String: " + modeStr);
+      }
+    }
+    return permissions;
+  }
+
+
+  /**
+   * Parses the permissions from the given octal mode value.
+   * 
+   * @param mode the octal permission mode
+   * @return the set of parsed permissions
+   */
+  public static Set<Permission> ofOctal(int mode) {
```

Comment:
Perhaps `fromOctal` is a better name?

Reply:
I have no problem to go with `fromModeOctal` if this is more suited.